lib/signs_ui/signs/request.ex

```diff
@@ -1,28 +1,21 @@
 defmodule SignsUI.Signs.Request do
   require Logger
-  alias SignsUI.Signs.Sign
+  alias SignsUI.Signs.{Sign, S3}
 
-  def get_signs(nil), do: %{}
-  def get_signs(url) do
-    http_client = Application.get_env(:signs_ui, :http_client)
-    case http_client.get(url) do
-      {:ok, %HTTPoison.Response{status_code: status, body: body}} when status >= 200 and status < 300 ->
-        parse(body)
-      {:ok, %HTTPoison.Response{status_code: status}} ->
-        Logger.warn("Could not load signs. Response returned with status code #{inspect status}")
-        %{}
-      {:error, %HTTPoison.Error{reason: reason}} ->
-        Logger.warn("Could not load signs: #{inspect reason}")
+  def get_signs() do
+    case S3.get_object() do
+      {:ok, %{body: json}} ->
+        parse(json)
+      {:error, reason} ->
+        Logger.warn("Could not load Signs config from AWS: #{inspect reason}")
```

Comment:
The app is kind of useless if this fails. (Just ran into this deploying this branch to AWS - had permissions wrong and was consistently shown a blank page with no signs).

I think it's better to log an error and then return some sort of error tuple here, so that the `GenServer.init` that calls this can return `{:stop, :failed_to_load_config}`.

That will in turn cause the app to fail to start, and work out a little better in the AWS environment. For instance, ElasticBeanstalk, when you deploy a new version, will boot up the new one in the background, and if all is well, switch to it. By making it fail to start up here, it will go on using the previous version (which presumably still works).

Reply:
Same in the other branch if it loads but fails to parse, I guess.